apollo-coroutines-support/src/main/kotlin/com/apollographql/apollo/coroutines/CoroutinesExtensions.kt

```diff
@@ -13,9 +13,8 @@ import kotlinx.coroutines.flow.*
 
 private class ChannelCallback<T>(val channel: Channel<Response<T>>) : ApolloCall.Callback<T>() {
 
-  @ExperimentalCoroutinesApi
   override fun onResponse(response: Response<T>) {
-    if (!channel.isClosedForSend) {
+    runCatching {
```

Comment:
@sav007 I feel the `runCatching` approach is safer since there could potentially be a race condition where the channel is closed just after the `if` before. Let me know if `isClosedForSend` was used to catch other cases.

Reply:
To avoid having to wrap the offer in a `runCatching`, you could create an extension on a Channel.

```
fun <E> SendChannel<E>.safeOffer(value: E) = runCatching {
        offer(value)
}

Reply:
@msya, for 2 instances only, I'd rather not pollute the `Channel` namespace with an extension function.